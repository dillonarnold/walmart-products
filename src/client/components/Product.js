import React from 'react';
import { shape, string, number } from 'react-proptypes';
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import Typography from '@material-ui/core/Typography';

const propTypes = {
  product: shape({
    itemId: number,
    name: string,
    mediumImage: string
  }).isRequired
};

const Product = ({product}) => (
  <Card id={`card-${product.itemId}`}>
    <CardContent>
      <Typography id={`cardName-${product.itemId}`} color="h2">
        {product.name}
      </Typography>
      <Typography id={`cardId-${product.itemId}`} color="textSecondary">
        {product.itemId}
      </Typography>
      <img id={`cardImage-${product.itemId}`} src={product.mediumImage} alt={product.name}/>
    </CardContent>
  </Card>
);

Product.propTypes = propTypes;

export default Product;
