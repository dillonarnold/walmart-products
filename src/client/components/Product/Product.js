import React from 'react';
import PropTypes from 'react-proptypes';
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import Typography from '@material-ui/core/Typography';

const propTypes = {
  /** Walmart product */
  product: PropTypes.shape({
    /** Id of the product */
    itemId: PropTypes.number,
    /** Name of the product */
    name: PropTypes.string,
    /** The medium image src of the product */
    mediumImage: PropTypes.string
  }).isRequired
};

/**
 * Displays a single Walmart product.
 *
 * @version 1.0.0
 * @author Dillon Arnold
 */
const Product = ({product}) => (
  <Card id={`card-${product.itemId}`}>
    <CardContent>
      <Typography id={`cardName-${product.itemId}`} color="h2">
        {product.name}
      </Typography>
      <Typography id={`cardId-${product.itemId}`} color="textSecondary">
        {product.itemId}
      </Typography>
      <img id={`cardImage-${product.itemId}`} src={product.mediumImage} alt={product.name}/>
    </CardContent>
  </Card>
);

Product.propTypes = propTypes;

export default Product;
