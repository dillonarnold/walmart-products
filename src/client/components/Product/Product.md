Example:

```jsx

const ipod = {
	itemId: 123456, 
	name: "iPod", 
	mediumImage: "https://i5.walmartimages.com/asr/cecd7151-08ae-47ce-84f2-b46abda0f5ca_1.22789781b5d7d5632eea50d3789f2156.jpeg?odnHeight=180&odnWidth=180&odnBg=FFFFFF"
};

<Product product={ipod} />
```
