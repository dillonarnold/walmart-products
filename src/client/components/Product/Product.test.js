import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Product from './Product';

describe('Product component', () => {
  const ipod = {
    itemId: 123456,
    name: 'iPod',
    mediumImage: 'imageSrc'
  };

  it('should render properly', () => {
    const component = shallow(
      <Product product={ipod} />,
    );

    expect(toJson(component)).toMatchSnapshot();

  });

  it('test props', () => {
    const component = mount(<Product product={ipod} />);
    expect(component.prop('product')).toEqual(ipod);
  });
});
