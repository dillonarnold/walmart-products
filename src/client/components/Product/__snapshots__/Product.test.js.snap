// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product component should render properly 1`] = `
<WithStyles(Card)
  id="card-123456"
>
  <WithStyles(CardContent)>
    <WithStyles(Typography)
      color="primary"
      id="cardName-123456"
    >
      iPod
    </WithStyles(Typography)>
    <WithStyles(Typography)
      color="textSecondary"
      id="cardId-123456"
    >
      123456
    </WithStyles(Typography)>
    <img
      alt="iPod"
      id="cardImage-123456"
      src="imageSrc"
    />
  </WithStyles(CardContent)>
</WithStyles(Card)>
`;
